__tests__/app.test.js

```diff
@@ -0,0 +1,89 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const testData = require("../db/data/test-data/index");
+const seed = require("../db/seeds/seed");
+const endpointsData = require("../endpoints.json");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("app", () => {
+  test("should return an object describing all available endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
+        console.log(response.body);
+        expect(response.body).toEqual(endpointsData);
+      });
+  });
+});
+
+describe("app", () => {
+  describe("get /api/topics", () => {
+    test("return request 200: returns status code", () => {
+      return request(app).get("/api/topics").expect(200);
+    });
+    test("return topics", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then((response) => {
+          const { topics } = response.body;
+          expect(Array.isArray(topics)).toBe(true);
+          expect(topics.length).toBe(3);
+          topics.forEach((topic) => {
+            expect(topic).toMatchObject({
+              slug: expect.any(String),
+              description: expect.any(String),
+            });
+          });
+        });
+    });
+  });
+});
+
+describe("app", () => {
+  describe("GET /api/articles/:article_id", () => {
+    test("return request 200: returns status code", () => {
+      const articleID = 1;
+      return request(app).get(`/api/articles/${articleID}`).expect(200);
+    });
+
+    test("should return an article by its id", () => {
+      const articleID = 1;
+      const expectedArticle = {
+        article_id: 1,
+        title: "Living in the shadow of a great man",
+        topic: "mitch",
+        author: "butter_bridge",
+        body: "I find this existence challenging",
+        created_at: "2020-07-09T20:11:00.000Z",
+        votes: 100,
+        article_img_url:
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+      };
+      return request(app)
+        .get(`/api/articles/${articleID}`)
+        .expect(200)
+        .then((response) => {
+          const { article } = response.body;
+          expect(article).toHaveProperty("author");
+          expect(article).toHaveProperty("title");
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("body");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("votes");
+          expect(article).toHaveProperty("article_img_url");
+          expect(article).toMatchObject(expectedArticle);
+        });
+    });
+  });
+});
```

Comment:
We are missing some error handling tests here, namely 400 for a bad request and 404 for an article not found